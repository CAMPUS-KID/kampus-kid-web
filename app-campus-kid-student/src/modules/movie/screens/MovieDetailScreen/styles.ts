import { css } from '@emotion/native';
import { Theme } from '@theme';

export const styleSheet = ({ colors, fontFamilies, fontSizes }: Theme) => ({
  errorText: css`
    color: ${colors.accent};
  `,
  headerImage: css`
    width: 100%;
    height: 300px;
  ` as any,
  titleText: css`
    color: ${colors.primary};
    font-family: ${fontFamilies.roobertBold};
    font-size: ${fontSizes.extraLarge};
    text-align: center;
  `
});
