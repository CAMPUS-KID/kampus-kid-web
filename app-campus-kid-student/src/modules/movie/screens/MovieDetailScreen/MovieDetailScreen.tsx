import { useFetchMovieDetail } from '@modules/movie/hooks';
import { useBuildStyles } from '@shared_hooks';
import { buildImageURL } from '@shared_utils';
import React from 'react';
import { Text, ActivityIndicator, View, Image } from 'react-native';
import { SafeAreaView } from 'react-native-safe-area-context';
import { styleSheet } from './styles';

export interface Props {
  route: {
    params: {
      movieId: number;
    };
  };
}

const MovieDetailScreen = ({
  route: {
    params: { movieId }
  }
}: Props) => {
  const styles = useBuildStyles(styleSheet);
  
  const { isFetchingMovieDetail, fetchMovieDetailError, movieDetail } =
    useFetchMovieDetail(movieId);

  if (isFetchingMovieDetail) {
    return (
      <SafeAreaView>
        <ActivityIndicator animating size="large" />
      </SafeAreaView>
    );
  }
  if (fetchMovieDetailError) {
    return (
      <SafeAreaView>
        <Text style={styles.errorText}>{fetchMovieDetailError.message}</Text>
      </SafeAreaView>
    );
  }
  return (
    <SafeAreaView>
      <View>
        <Image
          style={styles.headerImage}
          source={{ uri: buildImageURL(movieDetail.poster_path) }}
        />
        <Text style={styles.titleText}>{movieDetail.title}</Text>
      </View>
    </SafeAreaView>
  );
};

export default MovieDetailScreen;
