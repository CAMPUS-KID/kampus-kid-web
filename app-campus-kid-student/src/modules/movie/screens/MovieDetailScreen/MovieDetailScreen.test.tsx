import { wait } from '@test_shared';
import React from 'react';
import { render } from 'react-native-testing-library';
import { act } from 'react-test-renderer';
import MovieDetailScreen, { Props } from './MovieDetailScreen';

describe('MovieDetailScreen', () => {
  let props: Props;
  beforeEach(() => {
    props = {
      route: { params: { movieId: 100 } }
    };
  });
  it('should render', async () => {
    const { toJSON } = render(<MovieDetailScreen {...props} />);
    await act(() => wait());
    expect(toJSON()).toMatchSnapshot();
  });
});
