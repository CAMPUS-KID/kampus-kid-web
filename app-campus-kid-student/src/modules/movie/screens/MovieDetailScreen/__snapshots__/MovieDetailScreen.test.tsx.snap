// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetailScreen should render 1`] = `
<RNCSafeAreaView>
  <View>
    <Image
      source={
        Object {
          "uri": "undefined/1g0dhYtq4irTY1GPXvft6k4YLjm.jpg",
        }
      }
      style={
        Object {
          "height": 300,
          "width": "100%",
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#181818",
          "fontFamily": "Roobert-Bold",
          "fontSize": 28,
          "textAlign": "center",
        }
      }
    >
      Spider-Man: No Way Home
    </Text>
  </View>
</RNCSafeAreaView>
`;
